src/cipher/gadget.rs

```diff
@@ -0,0 +1,149 @@
+use super::{PoseidonCipher, CIPHER_SIZE, MESSAGE_CAPACITY};
+use dusk_plonk::constraint_system::ecc::Point;
+use dusk_plonk::prelude::*;
+use hades252::{GadgetStrategy, Strategy};
+
+/// Return a cipher provided the PoseidonCipher encryption parameters
+pub fn poseidon_cipher_gadget(
+    composer: &mut StandardComposer,
+    shared_secret: &Point,
+    nonce: Variable,
+    message: &[Variable],
+) -> [Variable; CIPHER_SIZE] {
+    let zero = composer.add_input(BlsScalar::zero());
```

Comment:
That has to be constrained to indeed be equal to zero.

There's a function to do that in plonk called: `add_constant_witness` which does it for you returning the `Variable` you need.

Reply:
Fixed